core/shared/src/main/scala/org/http4s/internal/Logger.scala

```diff
@@ -37,27 +39,40 @@ object Logger {
       message.headers.mkString("Headers(", ", ", ")", redactHeadersWhen)
     else ""
 
-  def defaultLogBody[F[_]: Concurrent](
+  def defaultLogBody[F[_]](
       message: Message[F]
-  )(logBody: Boolean): Option[F[String]] =
+  )(logBody: Boolean)(implicit F: Async[F]): Option[F[String]] =
     if (logBody) {
       val isBinary = message.contentType.exists(_.mediaType.binary)
       val isJson = message.contentType.exists(mT =>
         mT.mediaType == MediaType.application.json || mT.mediaType.subType.endsWith("+json")
       )
-      val bodyStream = if (!isBinary || isJson) {
-        message.bodyText
-      } else {
-        message.body.map(b => java.lang.Integer.toHexString(b & 0xff))
+
+      message.entity match {
+        case Entity.Empty =>
+          None
+        case Entity.Strict(bv) =>
+          if (!isBinary || isJson) {
+            val charset = message.charset.fold(Charset.`UTF-8`.nioCharset)(_.nioCharset)
+            Some(F.delay(bv.decodeStringLenient()(charset)))
```

Comment:
Sorry, I guess I'm not really following 😕 how is this an effect? There is no side-effect and it does not throw exception, right?

Reply:
> does not throw exception

It shouldn't throw anything if we 100% believe in fs2. But it's decoding... it feels like some things may happen.

Reply:
Even if it did throw an exception, we only need `ApplicativeError.catchNonFatal`.

Anyway, the method is pure. If it throws an exception in some situation, then we should fix it.

https://github.com/scodec/scodec-bits/blob/f9193339783694ed8523d6eb2ab000e73ff8a08d/core/shared/src/main/scala/scodec/bits/ByteVector.scala#L1089-L1106

Reply:
Oh Geez, I'm wrong about mentioning `fs2` here. It's from `scodec`.

Reply:
Alright, you've convinced me.